src/shadowbox/infrastructure/prometheus_scraper.ts

```diff
@@ -69,10 +69,14 @@ export class PrometheusClient {
   }
 }
 
-export async function runPrometheusScraper(
-    args: string[], configFilename: string, configJson: {},
-    prometheusEndpoint: string): Promise<child_process.ChildProcess> {
-  mkdirp.sync(path.dirname(configFilename));
+export async function startPrometheus(
+    configFilename: string, configJson: {}, processArgs: string[], endpoint: string) {
+  await writePrometheusConfigToDisk(configFilename, configJson);
```

Comment:
I wonder if it makes sense to make these two calls in main instead. It feels odd to pass both a config filename and the config contents to a method. The configJson is not really used by spawnPrometheusSubprocess.

Anyway, not important, up to you.

Reply:
I don't feel strongly either way about it.

The configFilename is used in the args, so there's kind of a duplication going on.

I would personally keep it as one method (even if the signature isn't that great) so that users don't forget to call writeConfigFile before spawnPromProcess.